api/task_manager.cc

```diff
@@ -183,6 +183,47 @@ void set_task_manager(http_context& ctx, routes& r) {
         auto s = co_await retrieve_status(task);
         co_return make_status(s);
     });
+
+    tm::get_task_status_recursively.set(r, [&ctx] (std::unique_ptr<request> req) -> future<json::json_return_type> {
+        auto& _ctx = ctx;
+        auto id = tasks::task_id{utils::UUID{req->param["task_id"]}};
+        std::queue<tasks::task_manager::foreign_task_ptr> q;
+        utils::chunked_vector<full_task_status> res;
+
+        // Get requested task.
+        auto task = co_await tasks::task_manager::invoke_on_task(_ctx.tm, id, std::function([] (tasks::task_manager::task_ptr task) -> future<tasks::task_manager::foreign_task_ptr> {
+            auto state = task->get_status().state;
+            if (state == tasks::task_manager::task_state::done || state == tasks::task_manager::task_state::failed) {
+                task->unregister_task();
+            }
```

Comment:
Why is a query for the task status has to take care of unregistering the task?

Reply:
Tasks are unregistered when they are finished (either successfully or not) and one of the following occurred:
- ttl time passed since the task has finished
- the task status was retrieved with an api.

Thus, it seemed reasonable to do it openly in the api call (as it is done for `get_task_status` and `wait_task`).
Maybe `task::get_status_and_unregister()`  should rather do that and be called from here?

Reply:
Also, I was wondering how `get_task_status_recursively` should behave regarding unregistration of tasks (unregister all of them/none/base task only?). I feel like it shouldn't do less than `get_task_status` as it is somehow its extension, but unregistering the whole tree seems to be a little to much (many calls to other shards). Do you agree?

Reply:
Sorry, I somehow missed your response.

I think it is odd that we remove the task after querying its status. I think we should always let it live until the TTL expires. But that is out of the scope of this PR.

Reply:
Won't unregistering the top level task also get rid of the other ones? Or they become orphaned until TTl expires?

Reply:
They are still stored in module until TTL expires or their status is requested.

Reply:
Ok, I think it is fine as-is for now, but as a follow-up I think we should consider removing this unregistering on status query. There might be more than one operators wanting to have a look at the outcome of a task, or one might want to look at it again. I think the current practice is confusing. @bhalevy what do you think?